Changes to support pcap2har on Google App Engine

mdwelsh opened the pull request:
These changes allow pcap2har to run cleanly on AppEngine, which is a great thing.

Also, this includes an earlier pull request for features that make pcap2har more robust in cases where there are missing packets in the trace.

Let me know what you think. Thanks!

Commit:
Merging in changes from Google:
  - Fix imports to allow pcap2har to be used as a library
  - Allow TCP packets to appear out of order
  - Ignore SDCH compression in HTTP responses

Commit:
Options added to pad missing TCP data (making HTTP parsing more
robust), and allowing for a mismatch in HTTP response sizes and the
Content-Length header.

Commit:
Adding README.mdw file

Commit:
Two small bugfixes:
  - Fix pad_missing_data when no chunks in a given flow
  - Allow ModifiedReader to read from a cStringIO object

Commit:
Removed dependency on socket.inet_ntoa

Commit:
A small set of changes to allow pcap2har to work under Google App
Engine, which requires Python 2.5. These changes make pcap2har
backwards compatible with Python2.5 and remove dependencies on
libraries not found in the App Engine SDK.

Commit:
Small bugfix to tcp/packet.py - using wrong class name in __ne__

andrewf commented:
Sorry it's taken me so long to get to this. School has been a bit rough.

I like a lot of the changes, especially padding incomplete TCP data. But I'm hesitant to include a lot of backwards-compatibility stuff in the mainline. I'm an idealist, and I want the code to be beautiful. :)

I didn't want to entangle dpkt_http_replacement with "regular" pcap2har code, but it might be the only way. dpkt doesn't seem to be maintained anymore. Maybe I should just fork the whole thing.

Also, I don't see where http_strict_parse_body ever gets set to False. Do you set that in your own code?

Commit:
fixed typo in log message

Commit:
added option to accept trailing semicolon in http media type

Commit:
Fixed incorrect array length check

Commit:
Merge pull request #1 from ethankb/minor_fixes

Minor fixes

Commit:
added command line option to expose settings.allow_trailing_semicolon

Commit:
Merge pull request #2 from ethankb/minor_fixes

Minor fixes - exposing the setting I added last time via command line

Commit:
added setting to handle empty content type and flag to set it

Commit:
Merge pull request #3 from ethankb/minor_fixes

Minor fixes - handling another malformed HTTP response from a crawl

Commit:
Removed a debugging variable that someone had left around that was never initialized.

Commit:
Fixed typo: log->logging

Commit:
Handle parsing problems for content-length, after observing '0,0' in a real trace.

Commit:
Better handling of parsing problems for content-length, after observing 0,0 in a real trace.

Commit:
Missing parantheses

Commit:
Fixed assert that seemed to have a logic bug in it.
The assert failed in cases when a new packet came in that merged onto
the back of an existing chunk and completely overlapped the next chunk.

Commit:
Merge pull request #4 from ethankb/minor_fixes

Minor fixes

andrewf commented:
Most of these have been fixed.
